lib/cargoNextest.nix

```diff
@@ -44,10 +42,8 @@ let
     '';
 
     checkPhaseCargoCommand = ''
-      cargo ${cargoExtraArgs} \
-        ${lib.optionalString withLlvmCov "llvm-cov ${cargoLlvmCovExtraArgs}"} \
-        nextest ${cmd} ''${CARGO_PROFILE:+--cargo-profile $CARGO_PROFILE} \
-          ${cargoNextestExtraArgs} ${moreArgs}
+      ${if withLlvmCov then "cargo llvm-cov nextest ${cargoLlvmCovExtraArgs}" else "cargo nextest ${cmd}"} \
+      ''${CARGO_PROFILE:+--cargo-profile $CARGO_PROFILE} ${cargoNextestExtraArgs} ${moreArgs}
```

Comment:
Is it possible/valid to bring back `${cargoExtraArgs}` but place them before `${cargoLLvmCovExtraArgs}`/`${cargoNextestExtraArgs}`?

The various subcommands all intentionally accept `cargoExtraArgs` to minimize the (re)configuration one might have to do between common and less common options. For example:

```nix
let
  commonArgs = {
    src = ...;
    cargoExtraArgs = "-p foo -p bar --features baz"; // Only build the specific crates/features for all operations
  };

  depsOnly = craneLib.buildDepsOnly commonArgs;

  tests = craneLib.cargoNextest (commonArgs // {
    cargoNextestExtraArgs = "--something --else"; // Don't need to repeat package or features here!
  });
in craneLib.buildPackage (commonArgs // {
  cargoArtifacts = tests;
});
```

Reply:
Yes, it would work for all configurations out there. And I think you're right. It doesn't have to be removed.

I can confirm that every cargo `Build Command` according to [The Cargo Book](https://doc.rust-lang.org/cargo/commands/build-commands.html) shares all of the `cargo build` command line options.

As long as the library user is aware (I wasn't at first), only to use the common `cargo build` options in `cargoExtraArgs`, it is viable to put it back.

On another note: Since nextest allows for `cargo test` command line options, should `craneLib.cargoNextest` also consume `cargoTestExtraArgs` in addition to `cargoExtraArgs`? That way all cargo-related options can be placed in something like `commonTestArgs` if anyone ever needs that.

Reply:
> On another note: Since nextest allows for `cargo test` command line options, should `craneLib.cargoNextest` also consume `cargoTestExtraArgs` in addition to `cargoExtraArgs`?

No I don't think we need to go that far. My (current) interpretation is that `cargoTestExtraArgs` would be for things specific to `craneLib.cargoTest`. I figure projects would want to use either `cargoTest` or `cargoNextest` but not both (though we can easily change this in the future so not too worried there)